ir/dict.go

```diff
@@ -142,6 +164,27 @@ func (d Dict) Get(key Node) Node {
 	return nil
 }
 
+func (d Dict) MarshalJSON() (b []byte, e error) {
+	// Temporal type to avoid recursion
+	type tmp Dict
+	ts := tmp(d)
+
+	return json.Marshal(&ts)
```

Comment:
Does this work? Where are you writing the "type" and "value" fields?

Reply:
This works "out-of-the-box". The fact that we implemented the MarshalJSON interface for every type allows json.Marshal to know how to Marshal Pairs without additional information. It works with all the tests that I've done, but I may be missing something.

Reply:
You were right here! I was missing the case where a `Dict` is part of another `Dict` inside a `Pair. I completely missed that case and that's why I thought it worked. Apologies! 

Will push the new version in a bit (we have a few meetings now and I want to revise it first).

Reply:
Now it is working. The test fails though because of this https://github.com/libp2p/go-smart-record/issues/5

Reply:
Tests pass. The issue was that in the test I had to pairs with the same `Key`